packages/kit/src/runtime/server/page/render.js

```diff
@@ -353,17 +354,32 @@ export async function render_response({
 	// add the content after the script/css links so the link elements are parsed first
 	head += rendered.head;
 
+	const html = options.app_template({
+		head,
+		body,
+		assets,
+		nonce: /** @type {string} */ (csp.nonce)
+	});
+
 	// TODO flush chunks as early as we can
-	const html =
+	const transformed =
 		(await resolve_opts.transformPageChunk({
-			html: options.app_template({ head, body, assets, nonce: /** @type {string} */ (csp.nonce) }),
+			html,
 			done: true
 		})) || '';
 
+	if (DEV) {
+		if (transformed.split('<!--').length < html.split('<!--').length) {
```

Comment:
something like `transformed.match(/<!--/g).length` might be less expensive as i think it wouldn't create large arrays (i'm not a regex expert - that might need some escaping or something)

Reply:
`String.prototype.match` called with a global regex _does_ return a regular array anyway (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match), so unless there's some clever optimization that works in one case but not the other, we're not really avoiding anything by doing that.

I don't think we want to worry overly much about performance here anyway, because it is a dev-only check,

Reply:
huh TIL about `match`. What a fantastically useless API! Struggling to imagine a single use case for it other than ours — counting — for which allocating an array is incredibly wasteful. What the hell am I supposed to do with the returned strings, without indexes etc?

It's `.match(string)` that really kills me though — it will return an array but it will only ever have a length of 1 and its value will only ever be `[input]`:

```js
// WHAT THE FUCK AM I SUPPOSED TO DO WITH THIS?!
'wtf'.match('wtf'); // ['wtf'] 
```

Anyway, this was a good hypothesis but I just ran some microbenchmarks...

```js
const html = `<!-- contents of view-source:https://kit.svelte.dev/ -->`

let iterations = 1e6;

function with_split() {
  let i = iterations;
  let count;
  console.time('with split');
  while (i--) {
    count = html.split('<!--').length;
  }
  console.timeEnd('with split');
}

function with_match() {
  let i = iterations;
  let count;
  console.time('with match');
  while (i--) {
    count = html.match(/<!--/g).length;
  }
  console.timeEnd('with match');
}

with_split();
with_match();
```

...and the `match` version is a full two orders of magnitude slower (7.4s vs 69ms — nice — on my machine).